src/components/SearchResults/index.jsx

```diff
@@ -1,112 +1,80 @@
 import { debounce } from 'lodash';
-import CustomSingleSelect from "../common/CustomSingleSelect";
 
-const { gray50, gray200, gray300, gray600 } = vars;
-const CustomViewButton = ({ view, listView, onClick, icon }) => (
-    <Button
-        sx={{
-            background: listView === view ? gray50 : 'transparent',
-            padding: '0.5rem 0.75rem',
-            border: `1px solid ${gray300}`,
-            '&.Mui-disabled': {
-                border: `1px solid ${gray300}` 
-            },
-            '& svg path': {
-                fill: listView !== view ? gray300 : 'currentColor'
-            }
-        }}
-        disabled={view === 'table' && true}
-        onClick={onClick}
-    >
-        {icon}
-    </Button>
-);
 const useMockApi = () => mockApi;
 
-const SearchResultsBox = () => {
-    const [numberOfVisiblePages, setNumberOfVisiblePages] = React.useState(20);
-    const [listView, setListView] = React.useState('list');
-    const [loading, setLoading] = useState(false)
+const SearchResults = () => {
+    const [loading, setLoading] = useState(false);
+    const [terms, setTerms] = useState({});
+    const [filters, setFilters] = useState([]);
+    const [checkedLabels, setCheckedLabels] = useState({});
     const query = useQuery();
-    const searchTerm = query.get('searchTerm');
-    const {  getMatchTerms } = useMockApi();
+    const { getMatchTerms } = useMockApi();
 
-    const [terms, setTerms] = React.useState({});
+    const searchTerm = query.get('searchTerm');
 
-    const handleNumberOfPagesChange = (v) => {
-        setNumberOfVisiblePages(v);
+    const handleCheckboxChange = (category, label) => {
+        setCheckedLabels((prev) => ({
+            ...prev,
+            [category]: {
+                ...prev[category],
+                [label]: !prev[category]?.[label]
+            }
+        }));
     };
-  
-  const fetchTerms = useRef(
-    debounce((searchTerm) => {
-      setLoading(true);
-      getMatchTerms(searchTerm, searchTerm)
-        .then((data) => {
-          const parsedData = termParser(data, searchTerm);
-          setTerms(parsedData);
-          setLoading(false);
-        })
-        .catch((error) => {
-          setLoading(false);
+
+    const fetchTerms = useRef(
+        debounce((searchTerm) => {
+            setLoading(true);
+            getMatchTerms(searchTerm, searchTerm)
+                .then((data) => {
+                    const parsedData = termParser(data, searchTerm);
+                    setTerms(parsedData);
+                    setFilters(parsedData.filters);
+                    setLoading(false);
+                })
+                .catch((error) => {
+                    setLoading(false);
+                });
+        }, 500)
+    ).current;
+
+    useEffect(() => {
+        fetchTerms(searchTerm);
+    }, [searchTerm, fetchTerms]);
+
+
+    const filterResults = (results, checkedLabels) => {
+        return results.filter(item => {
```

Comment:
@Aiga115 the getMatchTerms already accepts a filters object, we don't need to do this locally, double check with @ddelpiano and @jrmartin , thnx

Reply:
In this case, we don't need to call the endpoint with the filter.

The user can use the Search bar from any page/component, and the sidebar with filters will not always be present. In fact the only time the filters sidebar will be present is when the user arrives to the Search results page, and at this point we have the results already. 

For example, user is searching while not on the Results page:
User is in the Home page, user searches through the Search bar, and selects 'Select All' results. The results page will display all the results, and toggling the Filters on the sidebar should only filter out the results, but we don't need to call the endpoint again with the filter since we would have the results already (just not filtered) and it would an unnecessary operation.

The only time when it would make sense to call the endpoint with the filter, is if the user is in the Results page and performs another search. Since this is the exception and not the norm, I think the filtering should happen as is done in here.

cc @ddelpiano